bpo-40166: Change Unicode Howto so that it does not have a specific number of assigned code points.

amaajemyfren opened the pull request:
The link http://www.unicode.org/versions/latest/#Summary redirects to the summary section of the summary page of the latest version of the Unicode Standard. The current latest version is version 13.0.0 .

The last few versions of the page have had the following sort of Summary opening (latest standard)

![image](https://user-images.githubusercontent.com/32741226/78321707-64bdb880-7575-11ea-9560-e7a88f6d1cc5.png)

This could be better than writing the number 110 thousand characters which is currently not true.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
bpo-NNNN: Summary of the changes made
```

Where: bpo-NNNN refers to the issue number in the https://bugs.python.org.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `master`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNN)
```

Where: [X.Y] is the branch name, e.g. [3.6].

GH-NNNN refers to the PR number from `master`.

-->


<!-- issue-number: [bpo-40166](https://bugs.python.org/issue40166) -->
https://bugs.python.org/issue40166
<!-- /issue-number -->

amaajemyfren committed:
Update unicode.rst

Change the number of code points from a specific number to a link to the latest standard that has a description of how many code points there are.

amaajemyfren committed:
Update unicode.rst

Changed the link to the Unicode Consortium Technical site as the original link went to the new non technical site.

benjaminp commented on Doc/howto/unicode.rst:
```diff
@@ -156,7 +157,7 @@ UTF-8 has several convenient properties:
 References
 ----------
 
-The `Unicode Consortium site <https://www.unicode.org>`_ has character charts, a
```
What was wrong with this URL?

amaajemyfren replied:
This may be more nitpicky by me and could probably be removed from this PR. My reasoning is as follows:

This part of the [HOWTO](https://docs.python.org/3/howto/unicode.html#references) says:
![image](https://user-images.githubusercontent.com/32741226/78499485-a4dd9080-7759-11ea-8f20-267b1722decb.png)

To me
> character charts, a glossary, and PDF versions of the Unicode specification

suggests technical detail. The page at https://www.unicode.org redirects one to https://home.unicode.org/ which is more of a "_**marketing**_" than a "_**technical**_" page. Eventually after clicking a bit one can get to the technical page (scroll down and click on the "**_Unicode Technical Site_**" link). The technical link, https://www.unicode.org/main.html, is faster to the **Character Charts** (the page calls them code charts), **Glossary** and **Unicode Specifications** (at least in HTML. To get to the PDF version is 2 clicks from either page.)

benjaminp replied:
Let's not change this here.

amaajemyfren replied:
Ok.

I have reverted it and will resolve it. Please advise if you would like it to remain like this or would you be open to another PR that changes this url.

benjaminp commented on Doc/howto/unicode.rst:
```diff
@@ -41,8 +41,9 @@ but these are two different characters that have different meanings.
-0x10FFFF (about 1.1 million values, with some 110 thousand assigned so
-far).  In the standard and in this document, a code point is written
+0x10FFFF (about 1.1 million values, the actual number assigned is
+`less <http://www.unicode.org/versions/latest/#Summary>`_ than that).
```
I think the link should be on the words "actual number assigned" instead. Also, it should be `https://`.

amaajemyfren replied:
Ok. 

The issue is, though, that https://www.unicode.org/versions/latest/#Summary currently redirects to http://www.unicode.org/versions/Unicode13.0.0/#Summary without `https://`. I still think that using the [latest link](https://www.unicode.org/versions/latest/#Summary) is better since it does not need a change to the HOWTO with every other update of unicode (provided the Unicode consortium maintains their documentation format).

amaajemyfren committed:
Update unicode.rst

amaajemyfren committed:
Update unicode.rst

amaajemyfren committed:
Update unicode.rst

amaajemyfren committed:
Update unicode.rst

One test failed because of failure to download some libraries. I did a small change just to correct that.

amaajemyfren committed:
Update unicode.rst

amaajemyfren committed:
Update unicode.rst

amaajemyfren committed:
Update unicode.rst

amaajemyfren committed:
Refactor tests.

request-checks: false

amaajemyfren committed:
Update unicode.rst

amaajemyfren committed:
Update unicode.rst
